### PR TITLE
Fix polars 0 14 5 build

### PR DESCRIPTION
**Pull request by ghuls**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ghuls** [comment]
```
Setting up cross-python
+ case "${target_platform}" in
+ case "${target_platform}" in
+ maturin build --release --strip --target aarch64-apple-darwin --interpreter=/Users/runner/miniforge3/conda-bld/polars_1661275073942/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python
Finished setting up cross-python
/Users/runner/miniforge3/conda-bld/polars_1661275073942/work/conda_build.sh: line 24: /Users/runner/miniforge3/conda-bld/polars_1661275073942/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/maturin: Bad CPU type in executable
Traceback (most recent call last):
  File "/Users/runner/miniforge3/bin/conda-mambabuild", line 10, in <module>

```

I didn't see the actual error last time, but now I found the problem:
`maturin: Bad CPU type in executable`

**Comment by timkpaine** [comment]
looks like they've had some trouble before https://github.com/conda-forge/maturin-feedstock/issues/44

File: recipe/build.sh [hunk 1]
```diff
@@ -2,13 +2,25 @@
...
+esac
 
-maturin build --release --strip --manylinux off --interpreter="${PYTHON}"
+case "${target_platform}" in
+    'linux-64')
+        maturin build --release --strip --compatibility off --interpreter="${PYTHON}";;
+    'osx-64')
+        maturin build --release --strip --interpreter="${PYTHON}";;
+    'osx-arm64')
+        maturin build --release --strip --target aarch64-apple-darwin --interpreter="${PYTHON}";;
```
Comment (timkpaine) [inline recipe/build.sh:21]: this is a diff from before

**Comment by ghuls** [comment]
> looks like they've had some trouble before [conda-forge/maturin-feedstock#44](https://github.com/conda-forge/maturin-feedstock/issues/44)

At least it looks like it coulld be an arm64 binary:
```bash
$ wget https://anaconda.org/conda-forge/maturin/0.13.2/download/osx-arm64/maturin-0.13.2-py310h0def66e_0.tar.bz2

w$ tar xjf maturin-0.13.2-py310h0def66e_0.tar.bz2 

$ file bin/maturin 
bin/maturin: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE|HAS_TLV_DESCRIPTORS>
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,7 +19,8 @@ requirements:
     - cross-python_{{ target_platform }}  # [build_platform != target_platform]
     - crossenv                            # [build_platform != target_platform]
     - {{ compiler('c') }}
-    - {{ compiler('rust') }}
+    #- {{ compiler('rust') }}
```
Comment (isuruf) [inline recipe/meta.yaml:22]: You need this here as well

Comment (ghuls) [reply recipe/meta.yaml:22]: It worked much better without it:
```
requirements:
  build:
    - python                              # [build_platform != target_platform]
    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
    - crossenv                            # [build_platform != target_platform]
    - maturin >=0.13,<0.14                # [build_platform != target_platform]
    - {{ compiler('c') }}
    - {{ compiler('rust') }}
    - rust 1.65.0.dev20220819
    - posix                               # [win]
    - cmake
    - make                                # [unix]
    - cargo-bundle-licenses
```

```
...
Mamba failed to solve:
 - rust 1.65.0.dev20220819
 - cmake
 - make
 - gcc_linux-64 10.*
 - rust_linux-64
 - cargo-bundle-licenses

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package rust_linux-64-1.56.0.dev20210816-h383556e_0 requires rust 1.56.0.dev20210816.*, but none of the providers can be installed
- 
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 124, in mamba_get_install_actions
    solution = solver.solve_for_action(_specs, prefix)
  File "/opt/conda/lib/python3.9/site-packages/boa/core/solver.py", line 226, in solve_for_action
    t = self.solve(specs)
  File "/opt/conda/lib/python3.9/site-packages/boa/core/solver.py", line 216, in solve
    raise RuntimeError("Solver could not find solution." + error_string)
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - rust 1.65.0.dev20220819
 - cmake
 - make
 - gcc_linux-64 10.*
 - rust_linux-64
 - cargo-bundle-licenses

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package rust_linux-64-1.56.0.dev20210816-h383556e_0 requires rust 1.56.0.dev20210816.*, but none of the providers can be installed
- 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 239, in main
    call_conda_build(action, config)
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 211, in call_conda_build
    result = api.build(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 3091, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 2054, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 788, in expand_outputs
    for (output_dict, m) in deepcopy(_m).get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 2121, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(ref_metadata, conda_packages, pass_no=0,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 782, in finalize_outputs_pass
    fm = finalize_metadata(om, parent_metadata=parent_metadata,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 546, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(m,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 386, in add_upstream_pins
    build_deps, build_unsat, extra_run_specs_from_build = _read_upstream_pin_files(m, 'build',
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 373, in _read_upstream_pin_files
    deps, actions, unsat = get_env_dependencies(m, env, m.config.variant,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 138, in get_env_dependencies
    actions = environ.get_install_actions(tmpdir, tuple(dependencies), env,
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 126, in mamba_get_install_actions
    conflict_packages = parse_problems(str(e))
  File "/opt/conda/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 41, in parse_problems
    assert words[1] == "package", f"words = {repr(words)}"
AssertionError: words = ['-', '-', 'package', 'rust_linux-64-1.56.0.dev20210816-h383556e_0', 'requires', 'rust', '1.56.0.dev20210816.*,', 'but', 'none', 'of', 'the', 'providers', 'can', 'be', 'installed']
Traceback (most recent call last):
  File "./build-locally.py", line 105, in <module>
    main()
  File "./build-locally.py", line 93, in main
    run_docker_build(ns)
  File "./build-locally.py", line 29, in run_docker_build
    subprocess.check_call([script])
  File "/home/ghuls/software/anaconda3/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.

```

**Comment by ghuls** [comment]
> ```
> Setting up cross-python
> + case "${target_platform}" in
> + case "${target_platform}" in
> + maturin build --release --strip --target aarch64-apple-darwin --interpreter=/Users/runner/miniforge3/conda-bld/polars_1661275073942/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python
> Finished setting up cross-python
> /Users/runner/miniforge3/conda-bld/polars_1661275073942/work/conda_build.sh: line 24: /Users/runner/miniforge3/conda-bld/polars_1661275073942/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/maturin: Bad CPU type in executable
> Traceback (most recent call last):
>   File "/Users/runner/miniforge3/bin/conda-mambabuild", line 10, in <module>
> ```
> 
> I didn't see the actual error last time, but now I found the problem: `maturin: Bad CPU type in executable`

In the log I can see that `build_alias=x86_64-apple-darwin13.4.0` and `host_alias=arm64-apple-darwin20.0.0`.
Shouldn't the build alias be `build_alias=arm64-apple-darwin20.0.0`?

````
+SEGEDIT=arm64-apple-darwin20.0.0-segedit
+SEG_ADDR_TABLE=arm64-apple-darwin20.0.0-seg_addr_table
+SEG_HACK=arm64-apple-darwin20.0.0-seg_hack
+SIZE=arm64-apple-darwin20.0.0-size
+STRINGS=arm64-apple-darwin20.0.0-strings
+STRIP=arm64-apple-darwin20.0.0-strip
+_CONDA_PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_arm64_apple_darwin20_0_0
+ac_cv_func_malloc_0_nonnull=yes
+ac_cv_func_realloc_0_nonnull=yes
+build_alias=x86_64-apple-darwin13.4.0
+host_alias=arm64-apple-darwin20.0.0
Setting up cross-python
```

File: recipe/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/meta.yaml:22]: Without it cross compilation would not work. To fix that error, we need to update https://github.com/conda-forge/rust-activation-feedstock as well for 1.65.0

**Comment by ghuls** [comment]
Installing maturin oxs_64 when building osx_arm64 let it progress further. I guess if https://github.com/conda-forge/rust-activation-feedstock is updated for1.65.0, it will finally work.
```
+ case "${target_platform}" in
+ case "${target_platform}" in
+ maturin build --release --strip --target aarch64-apple-darwin --interpreter=/Users/runner/miniforge3/conda-bld/polars_1661290231868/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python
Finished setting up cross-python
📦 Including license file "LICENSE"
🍹 Building a mixed python/rust project
🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.7
🐍 Not using a specific python interpreter
   Compiling autocfg v1.1.0
   Compiling libc v0.2.132
   Compiling proc-macro2 v1.0.43
   Compiling unicode-ident v1.0.3
   Compiling quote v1.0.21
   Compiling syn v1.0.99
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
error: could not compile `cfg-if` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add aarch64-apple-darwin`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`


error[E0463]: can't find crate for `compiler_builtins`
```

File: recipe/meta.yaml (hunk 2 above)
Comment (ghuls) [reply recipe/meta.yaml:23]: Can  you make a nightly for the rust-activation-feedstock?

Comment (isuruf) [reply recipe/meta.yaml:23]: Can you send a PR?

Comment (ghuls) [reply recipe/meta.yaml:23]: It doesn't have an example for making it for a nightly version as far as I can see.

**Comment by xhochy** [comment]
See https://github.com/conda-forge/rust-activation-feedstock/pull/24

**Pull request closed without merging**